assets/src/edit-story/components/canvas/carousel/index.js

```diff
@@ -139,52 +145,64 @@ function Carousel() {
-					width="24"
-					height="24"
-					aria-label={ __( 'Scroll Left', 'amp' ) }
-				/>
-			</Area>
-			<List area="carousel" ref={ listRef } hasHorizontalOverflow={ hasHorizontalOverflow }>
-				{ pages.map( ( page, index ) => {
-					const isCurrentPage = index === currentPageIndex;
-					return (
-						<DropZone key={ index } onDrop={ onDrop } pageIndex={ index } >
-							<Page
-								key={ index }
-								draggable="true"
-								onClick={ handleClickPage( page ) }
-								onDragStart={ onDragStart( index ) }
-								isActive={ isCurrentPage }
-								ref={ ( el ) => {
-									pageRefs.current[ page.id ] = el;
-								} }
-								aria-label={ isCurrentPage ? sprintf( __( 'Page %s (current page)', 'amp' ), index + 1 ) : sprintf( __( 'Go to page %s', 'amp' ), index + 1 ) }
-							/>
-						</DropZone>
-					);
-				} ) }
-			</List>
-			<Area area="right-navigation">
-				<RightArrow
-					isHidden={ ! hasHorizontalOverflow }
-					onClick={ () => scrollBy( ( 2 * PAGE_WIDTH ) ) }
-					width="24"
-					height="24"
-					aria-label={ __( 'Scroll Right', 'amp' ) }
-				/>
-				<GridViewButton
-					isDisabled
-					width="24"
-					height="24"
-					aria-label={ __( 'Grid View', 'amp' ) }
-				/>
-			</Area>
-		</Wrapper>
+		<>
+			<Wrapper>
+				<Area area="left-navigation">
+					<LeftArrow
+						isHidden={ ! hasHorizontalOverflow }
+						onClick={ () => scrollBy( -( 2 * PAGE_WIDTH ) ) }
+						width="24"
+						height="24"
+						aria-label={ __( 'Scroll Left', 'amp' ) }
+					/>
+				</Area>
+				<List area="carousel" ref={ listRef } hasHorizontalOverflow={ hasHorizontalOverflow }>
+					{ pages.map( ( page, index ) => {
+						const isCurrentPage = index === currentPageIndex;
+						return (
+							<DropZone key={ index } onDrop={ onDrop } pageIndex={ index } >
+								<Page
+									key={ index }
+									draggable="true"
+									onClick={ handleClickPage( page ) }
+									onDragStart={ onDragStart( index ) }
+									isActive={ isCurrentPage }
+									ref={ ( el ) => {
+										pageRefs.current[ page.id ] = el;
+									} }
+									aria-label={ isCurrentPage ? sprintf( __( 'Page %s (current page)', 'amp' ), index + 1 ) : sprintf( __( 'Go to page %s', 'amp' ), index + 1 ) }
+								/>
+							</DropZone>
+						);
+					} ) }
+				</List>
+				<Area area="right-navigation">
+					<RightArrow
+						isHidden={ ! hasHorizontalOverflow }
+						onClick={ () => scrollBy( ( 2 * PAGE_WIDTH ) ) }
+						width="24"
+						height="24"
+						aria-label={ __( 'Scroll Right', 'amp' ) }
+					/>
+					<StyledGridViewButton
+						width="24"
+						height="24"
+						onClick={ openModal }
+						aria-label={ __( 'Grid View', 'amp' ) }
+					/>
+				</Area>
+			</Wrapper>
+			<Modal
+				isOpen={ isGridViewOpen }
+				onRequestClose={ closeModal }
+				contentLabel={ __( 'Grid View', 'amp' ) }
+			>
+				<Primary onClick={ closeModal }>
+					{ __( 'Back', 'amp' ) }
```

Comment:
`Modal` could be directly concerned with rendering the button and only receive the button label and use `onRequestClose` for `onClick`